### ISSUE TITLE
SmartScreen in Microsoft Defender is blocking SDA

### ISSUE THREAD

**Issue by yurasiner** [Jessecar96/SteamDesktopAuthenticator#922]
For some reason, Windows security does not advise running this application. This was not the case with previous updates, what could be the matter?

**Comment by yurasiner** [comment]
I'll clarify, I'm sure that I downloaded the update from the official source

**Comment by dyc3** [comment]
This happens all the time for newly released windows programs. As more people use SDA, eventually SmartScreen will no longer make a fuss about it. As long as you got SDA from this github repo, you're safe to ignore it.

**Comment by yurasiner** [comment]
> This happens all the time for newly released windows programs. As more people use SDA, eventually SmartScreen will no longer make a fuss about it. As long as you got SDA from this github repo, you're safe to ignore it.

I just downloaded again and the same problem

**Comment by dyc3** [comment]
Like I explained in my previous post, you're safe to ignore it. If you aren't comfortable with ignoring SmartScreen, then don't use SDA.

**Comment by saintlucifer1075** [comment]
can it be the reason that it do not log in and stuck on logging in screen?

**Comment by dyc3** [comment]
@saintlucifer1075  No. That would be an SDA bug (a known one, see other recent issues.)

**Comment by yurasiner** [comment]
> Like I explained in my previous post, you're safe to ignore it. If you aren't comfortable with ignoring SmartScreen, then don't use SDA.

On 1.12 I didn't have smart screen warning, and earlier too. If a lot of people have already downloaded it, why is there still a warning? And I can't stop using because there are no alternatives to it.

**Comment by dyc3** [comment]
> why is there still a warning?

Beats me. SmartScreen is somewhat of an enigma, and I don't use windows.

There _are_ alternatives. I maintain steamguard-cli. It's capable of reading the exactly the same maFiles that SDA does, so you don't have to deal with a trade ban. You can download it here: https://github.com/dyc3/steamguard-cli/releases

Read the quickstart here: https://github.com/dyc3/steamguard-cli/blob/master/docs/quickstart.md

Alternatively, some people have made community patches (#859), and there's other software that does the same thing, if you feel comfortable with that kind of thing. Otherwise, you can transfer the authenticator to the official steam mobile app, but that does cause a 2 day trade ban.

**Comment by yurasiner** [comment]
> you can transfer the authenticator to the official steam mobile app

I have a lot of accounts, I can't use them all on my phone

**Comment by yurasiner** [comment]
> > why is there still a warning?
> 
> Beats me. SmartScreen is somewhat of an enigma, and I don't use windows. As far as I know, SDA can't do anything about it.
> 
> There _are_ alternatives. I maintain steamguard-cli. It's capable of reading the exactly the same maFiles that SDA does, so you don't have to deal with a trade ban. You can download it here: https://github.com/dyc3/steamguard-cli/releases
> 
> Read the quickstart here: https://github.com/dyc3/steamguard-cli/blob/master/docs/quickstart.md
> 
> Alternatively, some people have made community patches (#859), and there's other software that does the same thing, if you feel comfortable with that kind of thing. Otherwise, you can transfer the authenticator to the official steam mobile app, but that does cause a 2 day trade ban.

Anyway. many thanks for the help!

**Comment by yurasiner** [comment]
> As more people use SDA, eventually SmartScreen will no longer make a fuss about it

in this page https://www.digitalcitizen.life/what-smartscreen-filter-how-does-it-work/ said: 

"The SmartScreen filter works by checking each website you visit and each file you try to download and run. It sends some quick information about them to Microsoft’s Windows Defender SmartScreen servers.

When Microsoft receives information from the SmartScreen filter about a particular website, it checks whether that website is on any list of known phishing or otherwise dangerous websites. If it is, it either warns you or blocks your access to it and lets you know that it’s not safe to continue."

Why is the source of this file dangerous in its database?

**Comment by LORD47** [comment]
> > you can transfer the authenticator to the official steam mobile app
> 
> I have a lot of accounts, I can't use them all on my phone


But the Steam app allows you to use multiple accounts on the same phone and these accounts don't need to be using the same phone number.

**Comment by LORD47** [comment]
> But i will have to change accounts every time to confirm

Well, if an extra 2 taps on screen is too much for you, maybe you should consider transfering to ASF as it comes with no Trade Hold at all and you can still keep using SDA alongside it too.

**Comment by yurasiner** [comment]
> Well, if an extra 2 taps on screen is too much for you, maybe you should consider transfering to ASF as it comes with no Trade Hold at all and you can still keep using SDA alongside it too.

These are not two additional clicks, hundreds of accounts and for each you will need to enter a password, and then confirm hundreds of confirmations

**Comment by LORD47** [comment]
> These are not two additional clicks, hundreds of accounts and for each you will need to enter a password, and then confirm hundreds of confirmations

First, you don't need to enter your password(s) each time, the Steam app supports adding and using multiple accounts on the same phone, but since you're talking about hundreds of accounts, then your best bet is to use [ASF](https://github.com/JustArchiNET/ArchiSteamFarm/wiki), it's even better than SDA in this regard and can confirm all your trades/market listing (for multiple accounts too) with a single command when you use its 2FA features.

PS: it's not advised to use more than 10 Steam accounts with the same IP.

**Comment by saintlucifer1075** [comment]
what is this ASF u talking about?

**Comment by LORD47** [comment]
> what is this ASF u talking about?

ArchiSteamFarm, an open source Steam Trading Cards farming tool that comes with other features, one of which is the 2FA  Module that plays thre role of an authenticator and when it's used, you can generate tokens and confirm/reject multiple trade/market listing and for more informations you may want to check its [Wiki](https://github.com/JustArchiNET/ArchiSteamFarm/wiki).

PS: you can import your *.maFile from SDA to ASF to enable the ASF 2FA and you won't get a Trade Hold at all and you get to keep using SDA alongside it.
